app/lib/features/news/widgets/news_item.dart

```diff
@@ -25,112 +26,145 @@ class NewsItem extends ConsumerWidget {
   });
 
   @override
-  Widget build(BuildContext context, WidgetRef ref) {
-    final slides = news.slides().toList();
+  ConsumerState<NewsItem> createState() => _NewsItemState();
+}
+
+class _NewsItemState extends ConsumerState<NewsItem> {
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addPostFrameCallback(
+      (_) {
+        ref.invalidate(newsSlideIndexProvider);
```

Comment:
It's already tagged as ```autoDispose()``` which means it will get reset upon riverpod drops listening of it, why its needed here?

Reply:
I need to reset horizontal newSlide index to 0 when user scroll up or down side. So it is required to `invalidate` or set state to 0.